src/components/subtitlesync/subtitlesync.js

```diff
@@ -1,170 +1,225 @@
+        this.subtitleSyncCloseButton.addEventListener('click', () => {
+            playbackManager.disableShowingSubtitleOffset(this.player);
+            this.toggle('forceToHide');
+        });
+    }
+
+    _tryShowSubtitleSync() {
+        // if showing subtitle sync is enabled and if there is an external subtitle stream enabled
+        if (!this._canShowSubtitleSync()) {
+            this.subtitleSyncContainer.classList.add('hide');
             return;
         }
 
-        if (player && playbackManager.supportSubtitleOffset(player)) {
-            if (!action) {
-                // if showing subtitle sync is enabled and if there is an external subtitle stream enabled
-                if (playbackManager.isShowingSubtitleOffsetEnabled(player) && playbackManager.canHandleOffsetOnCurrentSubtitle(player)) {
-                    // if no subtitle offset is defined or element has focus (offset being defined)
-                    if (!(playbackManager.getPlayerSubtitleOffset(player) || subtitleSyncTextField.hasFocus)) {
-                        // set default offset to '0' = 0ms
-                        subtitleSyncSlider.value = '0';
-                        subtitleSyncTextField.textContent = '0s';
-                        playbackManager.setSubtitleOffset(0, player);
-                    }
-                    // show subtitle sync
-                    subtitleSyncContainer.classList.remove('hide');
-                    return;
-                }
-            } else if (action === 'hide' && subtitleSyncTextField.hasFocus) {
-                // do not hide if element has focus
-                return;
-            }
-
-            subtitleSyncContainer.classList.add('hide');
+        // If no subtitle offset is defined or element has focus (offset being defined)
+        if (!(playbackManager.getPlayerSubtitleOffset(this.player) || this.subtitleSyncTextField.hasFocus)) {
+            // set default offset to '0' = 0ms
+            this.subtitleSyncSlider.value = '0';
+            this.subtitleSyncTextField.textContent = '0s';
+            playbackManager.setSubtitleOffset(0, this.player);
         }
-    }
 
-    update(offset) {
-        this.toggle();
+        // show subtitle sync
+        this.subtitleSyncContainer.classList.remove('hide');
+    }
 
-        const value = parseFloat(subtitleSyncSlider.value) + offset;
-        subtitleSyncSlider.updateOffset(value);
+    _canShowSubtitleSync() {
+        return playbackManager.isShowingSubtitleOffsetEnabled(this.player)
+               && playbackManager.canHandleOffsetOnCurrentSubtitle(this.player);
     }
 
     incrementOffset() {
-        this.update(+subtitleSyncSlider.step);
+        this.toggle();
+        this.offsetController.adjustOffset(+this.subtitleSyncSlider.step);
     }
 
     decrementOffset() {
-        this.update(-subtitleSyncSlider.step);
+        this.toggle();
+        this.offsetController.adjustOffset(-this.subtitleSyncSlider.step);
+    }
+
+    destroy() {
```

Comment:
Maybe move `destroy` and `toggle` right after `constructor` to keep the old sequence?
Or do you have any thoughts on their current positions?

Reply:
Nope, the old sequence sounds fine